lib/chef/resource/windows_certificate.rb

```diff
@@ -161,35 +161,69 @@ class WindowsCertificate < Chef::Resource
+
         def add_cert(cert_obj)
-          store = ::Win32::Certstore.open(new_resource.store_name)
+          store = ::Win32::Certstore.open(new_resource.store_name, store_location: get_cert_location[1])
           store.add(cert_obj)
         end
 
         def add_pfx_cert
           exportable = new_resource.exportable ? 1 : 0
-          store = ::Win32::Certstore.open(new_resource.store_name)
+          store = ::Win32::Certstore.open(new_resource.store_name, store_location: get_cert_location[1])
           store.add_pfx(new_resource.source, new_resource.pfx_password, exportable)
         end
 
         def delete_cert
           store = ::Win32::Certstore.open(new_resource.store_name)
-          store.delete(new_resource.source)
+          if check_string(new_resource.source) && new_resource.source.length == 40
+            store.delete(new_resource.source)
+          else
+            thumbprint = powershell_exec!(get_thumbprint(new_resource.store_name, get_cert_location[0], new_resource.source)).result
+            store.delete(thumbprint)
+          end
         end
 
         def fetch_cert
-          store = ::Win32::Certstore.open(new_resource.store_name)
-          store.get(new_resource.source)
+          store = ::Win32::Certstore.open(new_resource.store_name, store_location: get_cert_location[1])
+          if check_string(new_resource.source) && new_resource.source.length == 40
+            store.get(new_resource.source)
+          else
+            thumbprint = powershell_exec!(get_thumbprint(new_resource.store_name, get_cert_location[0], new_resource.source)).result
+            store.get(thumbprint)
+          end
+        end
+
+        # this Regex verifies whether a string is a hex value (ie. thumbprint) or not
+        def check_string(string)
+          string.scan(/\H/).empty?
+        end
+
+        def get_thumbprint(store_name, location, source)
+          <<-GETTHUMBPRINTCODE
+            $content = Get-ChildItem  -Path Cert:\\#{location}\\#{store_name} | Where-Object {$_.Subject -Match "#{source}"} | Select-Object Thumbprint
+            $content.thumbprint
+          GETTHUMBPRINTCODE
         end
 
         # Checks whether a certificate with the given thumbprint
         # is already present and valid in certificate store
         # If the certificate is not present, verify_cert returns a String: "Certificate not found"
         # But if it is present but expired, it returns a Boolean: false
         # Otherwise, it returns a Boolean: true
+        # updated this method to accept either a subject name or a thumbprint - 1/29/2021
+
         def verify_cert(thumbprint = new_resource.source)
-          store = ::Win32::Certstore.open(new_resource.store_name)
-          store.valid?(thumbprint)
+          store = ::Win32::Certstore.open(new_resource.store_name, store_location: get_cert_location[1])
+          status = check_string(thumbprint).to_s
+          if thumbprint.length == 40 && status == "true"
```

Comment:
that looks like you can `if check_string(thumbprint) && thumbprint.length == 40` here

Reply:
there we go, thanks